### PR TITLE
Added support for running migrations before rehydrating state

### PR DESCRIPTION
**Pull request by adam-golab**
Discussion about the desired feature can be found under the #4 issue.

TODO:
[X] add documentation to the readme

### CODE REVIEW

File: src/reduxPhoenix.js [hunk 1]
```diff
@@ -47,17 +84,24 @@ export default function persistStore(store, {
...
   throttle = 0,
+  migrations = null,
 } = {}) {
   return Promise.resolve(storage.getItem(key)).then(persistedJson => {
     if (disabled) {
       return store;
     }
     const persistedValue = deserialize(persistedJson);
-    const { persistedState, saveDate } = persistedValue || {};
+    const { persistedState, saveDate, migrations: appliedMigrations } = persistedValue || {};
```
Comment (ThibautMarechal) [inline src/reduxPhoenix.js:104]: You do not set "migrations" when saving into the storage

**Review by ThibautMarechal — CHANGES_REQUESTED** [review]
First of all thanks ! It worked well for my use case.

Maybe it we should add a validation for the migrations objects.

The code may throw in the migrations (the classic Cannot read property of '...' ) 
Maybe a try / catch around the migrations can catch the error ?

File: src/reduxPhoenix.js (hunk 1 above)
Comment (adam-golab) [reply src/reduxPhoenix.js:104]: It is saved. Please take a look on line 137

Comment (ThibautMarechal) [reply src/reduxPhoenix.js:104]: It was and old comment not submitted directly :)

Comment (adam-golab) [reply src/reduxPhoenix.js:104]: yeah, I forgot about it yesterday ;)

**Comment by adam-golab** [comment]
@ThibautMarechal good idea. I added some null checks a moment ago. I don't think that wrapping it in a `try/catch` block is necessary here. 

But I added a few checks that only migrations with the `name` and `up/down` method will be applied

**Review by ThibautMarechal — APPROVED** [review]

**Comment by adam-golab** [comment]
Published as a version: 1.3.0

**Pull request merged**